core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1154,9 +1152,9 @@ class ReplicaManager(val config: KafkaConfig,
    *  To avoid ISR thrashing, we only throttle a replica on the leader if it's in the throttled replica list,
    *  the quota is exceeded and the replica is not in sync.
    */
-  def shouldLeaderThrottle(quota: ReplicaQuota, topicPartition: TopicPartition, replicaId: Int): Boolean = {
-    val isReplicaInSync = nonOfflinePartition(topicPartition).exists(_.inSyncReplicaIds.contains(replicaId))
-    !isReplicaInSync && quota.isThrottled(topicPartition) && quota.isQuotaExceeded
+  def shouldLeaderThrottle(quota: ReplicaQuota, partition: Partition, replicaId: Int): Boolean = {
```

Comment:
Avoids a partition lookup in shouldLeaderThrottle

Reply:
Are the semantics the same? Do we ensure that `Partition` is not offline?

Reply:
On second thought, I think this doesn't matter.